src/Servers/Kestrel/Core/src/Internal/Http/Http1OutputProducer.cs

```diff
@@ -197,13 +213,20 @@ public void Advance(int bytes)
                     return;
                 }
 
-                if (_autoChunk)
+                if (!_startCalled)
                 {
-                    if (bytes < 0)
+                    if (bytes >= 0)
                     {
-                        throw new ArgumentOutOfRangeException(nameof(bytes));
-                    }
+                        if (_currentSegment.Length < _position + bytes)
+                        {
+                            throw new InvalidOperationException("Can't advance past buffer size.");
+                        }
 
+                        _position += bytes;
+                    }
+                }
+                else if (_autoChunk)
+                {
                     if (bytes + _advancedBytesForChunk > _currentChunkMemory.Length - BeginChunkLengthMax - EndChunkLength)
                     {
                         throw new InvalidOperationException("Can't advance past buffer size.");
```

Comment:
This should also be an ArgumentOutOfRangeException

Reply:
Ah this was very recent!

Reply:
https://github.com/dotnet/corefx/commit/498a7e169a452d3e53a061c1433aea637506a14a#diff-59ca508ec7e0ac1f99fa1b3f030f25ac